src/compiler/checker.rs

```diff
@@ -2493,37 +2637,175 @@ impl NodeBuilder {
             return factory
                 .create_literal_type_node(
-                    &synthetic_factory,
+                    &self.synthetic_factory,
                     &*Into::<Rc<Node>>::into(
                         if type_.as_intrinsic_type().intrinsic_name() == "true" {
-                            factory.create_true(&synthetic_factory)
+                            factory.create_true(&self.synthetic_factory)
                         } else {
-                            factory.create_false(&synthetic_factory)
+                            factory.create_false(&self.synthetic_factory)
                         },
                     ),
                 )
                 .into();
         }
+
+        let object_flags = get_object_flags(&*type_);
+
+        if type_.flags().intersects(TypeFlags::TypeParameter)
+            || object_flags.intersects(ObjectFlags::ClassOrInterface)
+        {
+            return if let Some(symbol) = type_.maybe_symbol() {
+                self.symbol_to_type_node(type_checker, symbol, context, SymbolFlags::Type)
+            } else {
+                unimplemented!()
+            };
+        }
+        if object_flags.intersects(ObjectFlags::Anonymous | ObjectFlags::Mapped) {
+            Debug_.assert(type_.flags().intersects(TypeFlags::Object), None);
+            return self.create_anonymous_type_node(type_checker, context, type_);
+        }
+
         unimplemented!()
     }
 
+    fn create_anonymous_type_node(
+        &self,
+        type_checker: &TypeChecker,
+        context: &NodeBuilderContext,
+        type_: Rc<Type /*ObjectType*/>,
+    ) -> TypeNode {
+        let symbol = type_.maybe_symbol();
+        if let Some(symbol) = symbol {
+            if false {
+                unimplemented!()
+            } else {
+                return self.visit_and_transform_type(
+                    type_checker,
+                    context,
+                    type_,
+                    NodeBuilder::create_type_node_from_object_type,
+                );
+            }
+        } else {
+            unimplemented!()
+        }
+    }
+
+    fn visit_and_transform_type(
+        &self,
+        type_checker: &TypeChecker,
+        context: &NodeBuilderContext,
+        type_: Rc<Type>,
+        transform: fn(&NodeBuilder, &TypeChecker, &NodeBuilderContext, Rc<Type>) -> TypeNode,
+    ) -> TypeNode {
+        let result = transform(self, type_checker, context, type_);
+        result
+    }
+
+    fn create_type_node_from_object_type(
+        &self,
+        type_checker: &TypeChecker,
+        context: &NodeBuilderContext,
+        type_: Rc<Type /*ObjectType*/>,
+    ) -> TypeNode {
+        let resolved = type_checker.resolve_structured_type_members(type_);
+
+        let members = self.create_type_nodes_from_resolved_type(type_checker, context, resolved);
+        let type_literal_node = factory.create_type_literal_node(&self.synthetic_factory, members);
```

Comment:
A "type literal" is the anonymous type that is associated with an object literal, I guess